packages/arb-token-bridge-ui/src/components/TransactionsTable/TransactionsTable.tsx

```diff
@@ -30,87 +45,244 @@ function EmptyTableRow({ children }: { children: React.ReactNode }) {
-              Failed to load transactions
+    <>
+      {/* top search and pagination bar */}
+      <div className="sticky top-0 left-0 flex w-auto flex-nowrap items-center justify-between gap-4 rounded-tr-lg bg-white p-3 text-sm">
+        {/* Search bar */}
+        <div className="relative flex h-full w-full grow items-center rounded border-2 bg-white px-2">
+          <SearchIcon className="h-4 w-4 shrink-0 text-gray-9" />
+          <input
+            className="text-normal h-full w-full p-2 font-light placeholder:text-gray-9"
+            type="text"
+            placeholder={`Search for ${layerType} transaction hash`}
+            value={searchString}
+            onChange={e => {
+              searchError ? setSearchError(false) : null
+              setSearchString(e.target.value)
+            }}
+            onKeyDown={e => {
+              if (e.key === 'Enter') {
+                search()
+              }
+            }}
+          />
+          {searchError ? (
+            <span className="absolute bottom-0 right-4 bg-white p-[9px] text-xs text-red-400">
+              {`Oops! Seems like a wrong ${layerType} transaction hash.`}
             </span>
-          </EmptyTableRow>
-        )}
-
-        {status === 'success' && (
-          <>
-            {transactions.length > 0 ? (
-              transactions.map((tx, index) => {
-                const isFinalRow = index === transactions.length - 1
-
-                return isDeposit(tx) ? (
-                  <TransactionsTableDepositRow
-                    key={`${tx.txId}-${tx.direction}`}
-                    tx={tx}
-                    className={!isFinalRow ? 'border-b border-gray-10' : ''}
-                  />
-                ) : (
-                  <TransactionsTableWithdrawalRow
-                    key={`${tx.txId}-${tx.direction}`}
-                    tx={tx}
-                    className={!isFinalRow ? 'border-b border-gray-10' : ''}
-                  />
-                )
-              })
-            ) : (
-              <EmptyTableRow>
+          ) : null}
+        </div>
+        {/* Pagination buttons */}
+        <div className="flex  w-auto  shrink grow-0 flex-row flex-nowrap items-center justify-end text-gray-10">
+          <button
+            className="rounded border border-gray-10 p-1"
+            onClick={prev}
+            disabled={!pageParams?.pageNumber}
+          >
+            <ChevronLeftIcon className="h-3 w-3" />
+          </button>
+
+          <div className="whitespace-nowrap p-2">
+            Page {(pageParams?.pageNumber || 0) + 1}{' '}
+          </div>
+
+          <button className="rounded border border-gray-10 p-1" onClick={next}>
+            <ChevronRightIcon className="h-3 w-3" />
+          </button>
+        </div>
+      </div>
+
+      {/* when there are no search results found */}
+      {status === 'success' && !transactions.length && searchString && (
```

Comment:
very nitpicky but do we want to run `.trim()` on `searchString`. If a user fatfingers space at the start or the end they have to manually remove it, can be annoying 😄

Reply:
Good suggestion. Added trim() in search and now it is working with spaces as well :)